openedx/core/djangoapps/credit/views.py

```diff
@@ -405,3 +406,60 @@ def get_object(self):
             self.kwargs[self.lookup_field] = CourseKey.from_string(course_key)
 
         return super(CreditCourseViewSet, self).get_object()
+
+
+class CreditProviderViewSet(viewsets.ReadOnlyModelViewSet):
+    """ CreditProvider endpoints. """
+
+    lookup_field = 'provider_id'
+    lookup_url_kwarg = 'provider_id'
+    queryset = CreditProvider.objects.all()
+    serializer_class = CreditProviderSerializer
+    authentication_classes = (OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
+    pagination_class = None
+
+    # This CSRF exemption only applies when authenticating without SessionAuthentication.
+    # SessionAuthentication will enforce CSRF protection.
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super(CreditProviderViewSet, self).dispatch(request, *args, **kwargs)
+
+    def list(self, request):
+        provider_id = request.GET.get("provider_id", None)
+        providers_list = provider_id.split(",") if provider_id else None
+        providers = CreditProvider.get_credit_providers(providers_list)
+
+        return Response(providers)
+
+    def retrieve(self, request, provider_id=None):
+        provider = CreditProvider.get_credit_provider(provider_id)
+
+        return Response(provider)
+
+
+class CreditEligibilityViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    """CreditEligibility endpoints. """
+
+    queryset = CreditEligibility.objects.all()
+    serializer_class = CreditEligibilitySerializer
+    authentication_classes = (OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
```

Comment:
`IsAdminUser` won't work where. You need a permissions class that allows the owner/user to view/create data.